apps/api/src/app/tenant/tenant.controller.ts

```diff
@@ -11,14 +20,34 @@ import { CreateTenantCommand } from './usecases/create-tenant/create-tenant.comm
 import { CreateTenantRequestDto } from './dtos/create-tenant-request.dto';
 import { ExternalApiAccessible } from '../auth/framework/external-api.decorator';
 import { ApiResponse } from '../shared/framework/response.decorator';
+import { GetTenantResponseDto } from './dtos/get-tenant.response';
+import { GetTenant } from './usecases/get-tenant/get-tenant.usecase';
+import { GetTenantCommand } from './usecases/get-tenant/get-tenant.command';
 
 @Controller('/tenants')
 @ApiTags('Tenants')
 @UseInterceptors(ClassSerializerInterceptor)
 @UseGuards(JwtAuthGuard)
 @ApiExcludeController()
 export class TenantController {
-  constructor(private CreateTenantUsecase: CreateTenant) {}
+  constructor(private CreateTenantUsecase: CreateTenant, private getTenantUsecase: GetTenant) {}
+
+  @Get('/:tenantId')
+  @ApiResponse(GetTenantResponseDto)
+  @ApiOperation({
+    summary: 'Get tenant',
+    description: `Get tenant by your internal id used to identify the tenant`,
+  })
+  @ExternalApiAccessible()
+  getTenantById(@UserSession() user: IJwtPayload, @Param('tenantId') tenantId: string): Promise<GetTenantResponseDto> {
```

Comment:
@djabarovgeorge I have a doubt. here`tenantId` param can be any string or MongoDB id?

Reply:
This one should be Novu internal id so MongoDB Id. If the user will provide not MongoDB id. 
MongoDB will throw an exception. do you think we should add the validation on the controller level? (actually, I did not think about it because I have not seen yet that we make such validation on query params)

Reply:
yes we can create a dto for this and in dto use `@IsMongo()` decorator for this field

Reply:
After some discovery, we decided to use here the tenant identifier, so the user won't need to manage/work with Novu's internal ids.
So I will refactor it to the tenant identifier which is a string.